internal/auth/middleware.go

```diff
@@ -0,0 +1,79 @@
+// Copyright © 2018 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/banzaicloud/pipeline/auth"
+	"github.com/gin-gonic/gin"
+)
+
+type enforcer interface {
+	Enforce(rvals ...interface{}) bool
+}
+
+// NewMiddleware returns a new gin middleware that checks user authorization.
+func NewMiddleware(e enforcer, basePath string) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		a := &middleware{
```

Comment:
This could be created outside the anonymous function in `NewMiddleware` and captured.

Reply:
Good catch